### PR TITLE
[WIP] adding download button to public gallery

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

File: templates/public.php [hunk 1]
```diff
@@ -6,7 +7,12 @@
 			<img class="svg" src="<?php print_unescaped(image_path('', 'logo-wide.svg')); ?>" alt="<?php p($theme->getName()); ?>" /></a>
 		<div id="logo-claim" style="display:none;"><?php p($theme->getLogoClaim()); ?></div>
 		<div class="header-right">
-			<span id="details"><?php p($l->t('shared by %s', $_['displayName'])) ?></span>
```
Comment (LukasReschke) [inline templates/public.php:11]: If we don't need the `displayName` here anymore we should also remove the assignment in `gallery/public.php`

**Comment by nikita81** [comment]
hi there,
I create the new public.php under apps/gallery/templates as described here, but I do not see any download button. I have owncloud 6.
Could you help please?
I logged in Owncloud, opened a gallery and the pictures inside, right, left, x and slide button are there but not download.

**Comment by jancborchardt** [comment]
@DeepDiver1975 @LukasReschke can you remove the assignment as noted above?

@owncloud/designers cc

**Comment by jancborchardt** [comment]
@DeepDiver1975 @LukasReschke can you remove the assignment as noted above?

Would be good if we can backport this to 7.0.1

**Comment by MorrisJobke** [comment]
@jancborchardt Just do it ;)

File: templates/public.php (hunk 1 above)
Comment (MorrisJobke) [inline templates/public.php:11]: FIxed

**Comment by jancborchardt** [comment]
Just tested – doesn’t seem to work unfortunately. :(

@icewind1991 can you look into fixing this? Would be good to have in 7.0.1 or 7.0.2

**Comment by DeepDiver1975** [comment]
rebased - further development blocked because of https://github.com/owncloud/core/issues/11312

**Comment by jospoortvliet** [comment]
@DeepDiver1975 that's fixed now with owncloud/core#11029 then? ;-)

**Comment by oparoz** [comment]
It's been implemented in the Gallery rewrite.

You can see what it looks like in the [demo](https://oc7demo.interfacloud.com/index.php/apps/galleryplus/public?token=312b038d3e70877199a4d66abf37bdd0) and the [source code for OC7](https://github.com/interfasys/galleryplus/tree/stable7) is now finally available.

**Comment by jancborchardt** [comment]
@oparoz is this based on the current master of the Gallery app? Because there are important accessibility features and other fixes that we need to have in.

It would also be good to not have a complete »rewrite« but to gradually apply changes to the existing app via Pull Requests. Especially as the app looks basically the same and does the exact same thing.

**Comment by oparoz** [comment]
Yes, current master with a few additions to make it compatible
with the current install base. So, nothing is lost in terms of usability.
I've only added the PR which were waiting to be rebased or merged. The
Javascript organisation is the same. Only the ajax endpoints have changed.

You can't really gradually port the app over to the AppFramework. It's a
different approach and we can finally get rid of most calls to the private
API (even more so on OC8).
The PHP architecture needs to be reviewed to make sure it's easily
understandable and manageable and then that can be used as a new foundation
to improve performance and implement new features.

**Comment by jancborchardt** [comment]
@oparoz ah ok – so this app is intended to replace the current Gallery app then, right?

**Comment by oparoz** [comment]
```
       Yes. I should now have a good understanding of both the
```

AppFramework and the Gallery app to be able to take it where you guys want.
If I disagree with where you want to go, I can always fork it then, but I'd
rather not maintain a separate branch.

**Comment by jancborchardt** [comment]
Yeah, we’d rather not do that either. :) I believe we’ll work it out as we all have the same goal: Make a great Pictures app for ownCloud users!

**Comment by DeepDiver1975** [comment]
I'm closing this as any further effort should go into galleryplus by @oparoz

**Comment by jancborchardt** [comment]
@oparoz is this functionality already present in Gallery+?

**Comment by oparoz** [comment]
Yes

**Comment by jancborchardt** [comment]
Awesome :)

**Pull request closed without merging**